hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/NearCacheProvider.java

```diff
@@ -79,46 +70,47 @@ private boolean isBatchingEnabled() {
         return hazelcastProperties.getBoolean(MAP_INVALIDATION_MESSAGE_BATCH_ENABLED) && batchSize > 1;
     }
 
-    public NearCache<Data, Object> getOrCreateNearCache(String mapName) {
-        return getOrPutIfAbsent(nearCacheMap, mapName, nearCacheConstructor);
+    public NearCache getOrCreateNearCache(String mapName) {
+        NearCacheConfig nearCacheConfig = nodeEngine.getConfig().getMapConfig(mapName).getNearCacheConfig();
+        EvictionConfigAccessor.initDefaultMaxSize(nearCacheConfig.getEvictionConfig());
+
+        NearCacheContext nearCacheContext = new NearCacheContext(
+                nearCacheManager,
+                nodeEngine.getSerializationService(),
+                new MemberNearCacheExecutor(nodeEngine.getExecutionService()),
+                null);
```

Comment:
I copied this from the `AbstractClientInternalCacheProxy`. Actually the only usage of that parameter seems to be in Enterprise.